packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -0,0 +1,337 @@
+    { src: C, dest: '@Ethereum', amount: token(70n) },
+    { src: '@Ethereum', dest: '@noble', amount: token(70n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(90n) },
+    { src: '@Arbitrum', dest: A, amount: token(90n) },
+  ]);
+});
+
+test('solver distribute from one (A -> B 60, A -> C 40)', async t => {
+  const current = balances({ [A]: 100n, [B]: 0n, [C]: 0n });
+  const target = { [A]: ZERO, [B]: token(60n), [C]: token(40n) };
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target,
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: A, dest: '@Arbitrum', amount: token(100n) },
+    { src: '@Arbitrum', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(60n) },
+    { src: '@noble', dest: '@Ethereum', amount: token(40n) },
+    { src: '@Avalanche', dest: B, amount: token(60n) },
+    { src: '@Ethereum', dest: C, amount: token(40n) },
+  ]);
+});
+
+test('solver collect to one (B 30 + C 70 -> A)', async t => {
+  const current = balances({ [A]: 0n, [B]: 30n, [C]: 70n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { [A]: token(100n), [B]: ZERO, [C]: ZERO },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: B, dest: '@Avalanche', amount: token(30n) },
+    { src: '@Avalanche', dest: '@noble', amount: token(30n) },
+    { src: C, dest: '@Ethereum', amount: token(70n) },
+    { src: '@Ethereum', dest: '@noble', amount: token(70n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(100n) },
+    { src: '@Arbitrum', dest: A, amount: token(100n) },
+  ]);
+});
+
+test('solver deposit redistribution (+agoric 100 -> A 70, B 30)', async t => {
+  const current = balances({ '+agoric': 100n, [A]: 0n, [B]: 0n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { '+agoric': ZERO, [A]: token(70n), [B]: token(30n) },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: '+agoric', dest: '@agoric', amount: token(100n) },
+    { src: '@agoric', dest: '@noble', amount: token(100n) },
+    { src: '@noble', dest: '@Arbitrum', amount: token(70n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(30n) },
+    { src: '@Arbitrum', dest: A, amount: token(70n) },
+    { src: '@Avalanche', dest: B, amount: token(30n) },
+  ]);
+});
+
+test('solver deposit redistribution (Deposit 100 -> A 70, B 30)', async t => {
+  const current = balances({ '<Deposit>': 100n, [A]: 0n, [B]: 0n });
+  const { steps } = await planRebalanceFlow({
+    mode: 'cheapest',
+    network: TEST_NETWORK,
+    current,
+    target: { '<Deposit>': ZERO, [A]: token(70n), [B]: token(30n) },
+    brand: TOK_BRAND,
+  });
+  t.deepEqual(steps, [
+    { src: '<Deposit>', dest: '@agoric', amount: token(100n) },
+    // TODO dckc should this  go through +agoric?
```

Comment:
I think the test case at line 142 will suffice and we don't need this test case anymore- the contract will handle moving the tokens from `<Deposit>` to `+agoric`